### PR TITLE
move decode_helpers to std/private

### PR DESCRIPTION
**Pull request by ringabout**
`handleHexChar` is very common to use in stdlib, let's export it!

compiler/lexer
```nim
proc handleHexChar(L: var Lexer, xi: var int; position: range[0..4])
```

lib/pure/parsecfg
```nim
proc handleHexChar(c: var CfgParser, xi: var int)
```

lib/pure/parsejson
```nim
proc handleHexChar(c: char, x: var int): bool
```

lib/pure/parsesql
```nim
proc handleHexChar(c: var SqlLexer, xi: var int)
```

lib/pure/pegs
```nim
proc handleHexChar(c: var PegLexer, xi: var int)
```

nimsuggest/sexps
```nim
proc handleHexChar(c: char, x: var int): bool
```

And uri and cgi module. They all share similar parts and could use `decode_helpers` instead!!!

- [ ] If this PR is acceptable, next PR I will clean them all.

### CODE REVIEW

File: lib/std/decode_helpers.nim
Comment (timotheecour) [inline lib/std/decode_helpers.nim]: s/string/openArray[char]/

File: lib/std/decode_helpers.nim
Comment (timotheecour) [inline lib/std/decode_helpers.nim]: pre-existing but now that it's exported, it needs some cleanup:
s/charracter/character/

File: lib/std/decode_helpers.nim
Comment (timotheecour) [inline lib/std/decode_helpers.nim]: lexer.handleHexChar diverges here:
```nim
  of '"', '\'':
    if position <= 1: invalid()
    # do not progress the bufpos here.
    if position == 0: inc(L.bufpos)
```
how will this be reconciled?

File: lib/std/decode_helpers.nim
Comment (timotheecour) [inline lib/std/decode_helpers.nim]: rename it to `lib/std/private/decode_helpers.nim`, that way it discourages people to import this and buys more time to "fix the API" if needed; we can always later move it to  `std/decode_helpers.nim` (or some other name) when this stabilizes.

Also, that way, you don't need a changelog entry since it's importable but still considered private (still much better than include and much better than code duplication as is case before your PR).

Comment (ringabout) [reply lib/std/decode_helpers.nim]: I agree.

File: lib/std/decode_helpers.nim
Comment (ringabout) [reply lib/std/decode_helpers.nim]: I am not sure.
maybe
```nim
case c:
of '0' .. '9', 'a' .. 'f': handleHexChar(c)
else: ...
```
Or don't change that

**Review by timotheecour — APPROVED** [review]

**Comment by Araq** [comment]
This PR is fine but (ignoring the special rules for Araq) a PR needs to have approval from two people before merging it.

**Pull request merged**
